### PR TITLE
Improve contribution guidelines

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*
This PR aims to improve the contributing guidelines for d2l, bringing all the details under one single roof i.e. `CONTRIBUTING.md`.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by AnirudhDagar** [comment]
@astonzhang feel free to edit/share your feedback on this PR. I've left it under draft for now.

**Comment by AnirudhDagar** [comment]
@d2l-bot please rebuild this!

**Comment by astonzhang** [comment]
1. Please also mention https://d2l.ai/chapter_appendix-tools-for-deep-learning/contributing.html
2. For installation, please refer to our installation page. Otherwise we'll soon find inconsistency among framework versions in different sections (e.g., in this PR it says pip install torch==1.8.1 torchvision==0.9.1, among other things)

**Comment by AnirudhDagar** [comment]
I've added the references. But installation section doesn't specify the exact version of the frameworks installed. I was thinking it would be nice to have this single CONTRIBUTING.md doc (everything under a single roof) as the one which we can always point new contributors to.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,10 +40,163 @@ For contributions that will take a significant amount of time, [open a new issue
...
+
+```bash
+conda activate d2l
+
+# PyTorch
+pip install torch==<version> torchvision==<version>
+# pip install torch==1.11.0 torchvision==0.12.0
+
+# MXNet
+pip install mxnet==<version>
```
Comment (cheungdaven) [inline CONTRIBUTING.md:89]: what about GPU versions?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -39,10 +40,163 @@ For contributions that will take a significant amount of time, [open a new issue
...
+# pip install mxnet==1.7.0
+
+# Tensorflow
+pip install tensorflow==<version> tensorflow-probability==<version>
+# pip install tensorflow==2.8.0 tensorflow-probability==0.16.0
+```
+
+Development of the book is powered by the
```
Comment (cheungdaven) [inline CONTRIBUTING.md:97]: Compilation

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -39,10 +40,163 @@ For contributions that will take a significant amount of time, [open a new issue
...
+```
+
+Now you can use `from d2l import <framework_name> as d2l` within the
+environment to access the saved functions and also edit them on the fly.
+
+When adding a code cell from a specific framework, one needs to specify
+the framework by commenting the following on top of a cell: `#@tab tensorflow`
+for example. If the code tab is exactly the same for all frameworks then
+use `#@tab all`. This information is required by the `d2lbook` package to
+build the website, pdf etc. We recommend looking at some of the notebooks
```
Comment (cheungdaven) [inline CONTRIBUTING.md:122]: pdf,

File: CONTRIBUTING.md (hunk 1 above)
Comment (AnirudhDagar) [reply CONTRIBUTING.md:89]: Fixed

**Comment by AnirudhDagar** [comment]
@astonzhang @cheungdaven this is ready, can we get this in?

**Comment by astonzhang** [comment]
@AnirudhDagar  Thanks for the great work!

**Pull request merged**